app/components/document_component.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Inherit from Arclight::DocumentComponent to create a custom document component
+# Conditionally render different metadata depending on component level
+class DocumentComponent < Arclight::DocumentComponent
+  # rubocop:disable Metrics/MethodLength
+  def document_body_component
```

Comment:
This could be dried up a bit if you make the initialize method signatures the same for `DocumentBodyCollectionComponent` and `DocumentBodyComponent`. Then you could have separate methods in this component that determine the right class and the right set of partials.